Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/ExpLog.lean

```diff
@@ -74,14 +75,21 @@ end RCLikeNormed
 
 section RealNormed
 
-variable {A : Type*} {p : A → Prop} [NormedRing A] [StarRing A]
+variable {A : Type*} [NormedRing A] [StarRing A]
   [TopologicalRing A] [NormedAlgebra ℝ A] [CompleteSpace A]
-  [ContinuousFunctionalCalculus ℝ p]
+  [ContinuousFunctionalCalculus ℝ (IsSelfAdjoint : A → Prop)]
 
-lemma real_exp_eq_normedSpace_exp {a : A} (ha : p a := by cfc_tac) :
+lemma real_exp_eq_normedSpace_exp {a : A} (ha : IsSelfAdjoint a := by cfc_tac) :
     cfc Real.exp a = exp ℝ a :=
   Real.exp_eq_exp_ℝ ▸ exp_eq_normedSpace_exp ha
 
+@[aesop safe apply (rule_sets := [CFC])]
+lemma _root_.NormedSpace.exp_nonneg_of_isSelfAdjoint {𝕜 : Type*} [Field 𝕜] [Algebra 𝕜 A]
+    [PartialOrder A] [StarOrderedRing A] {a : A} (ha : IsSelfAdjoint a := by cfc_tac) :
+    0 ≤ exp 𝕜 a := by
```

Comment:
I'm not sure I agree about point 3 here: sure, `cfc_tac` has a name that sounds like it's exclusively about the CFC, but in practice it should be useful for operator algebras more generally. And in this particular case, from the user's point of view, the fact that `exp` is not defined via the CFC is really an implementation detail. (And I agree about point 4, for exactly that reason!)

Reply:
I guess I'm relatively hesitant to use auto-params in lots of places, especially ones that invoke `aesop`. I feel like it could easily lead to performance issues (and already has on a few occasions).

In the continuous functional calculus we need it because those hypotheses are *everywhere*, and we also know that for the most part, it should be relatively easy to derive them from the context. But for this lemma, that's not so clear.

On the other hand, if in the future we rewrite (and probably rename) `cfc_tac` so that it is relatively reliable and fast (and therefore likely doesn't depend on `aesop`; or else depends on a "pruned" `aesop`), then I would be in favor of adding the appropriate tag more liberally across the library.

Does this make sense? Do you still disagree?

Reply:
No, that's fair enough.